app/ui-react/syndesis/src/modules/integrations/components/IntegrationEditorStepAdder.tsx

```diff
@@ -45,74 +60,145 @@ export interface IIntegrationEditorStepAdderProps {
-              s.action.descriptor.outputDataShape;
-            const shape =
-              idx === 0
-                ? getDataShapeText(s.stepKind!, outputDataShape)
-                : getDataShapeText(s.stepKind!, inputDataShape);
-
-            return (
-              <React.Fragment key={idx}>
-                <IntegrationEditorStepsListItem
-                  stepName={(s.action && s.action!.name) || s.name!}
-                  stepDescription={(s.action! && s.action!.description) || ''}
-                  additionalInfo={[
-                    <div
-                      className={
-                        'list-view-pf-additional-info-item-stacked list-view-pf-additional-info-item'
-                      }
-                      key={0}
-                    >
-                      <strong>{shape || 'n/a'}</strong>
-                      <span>Data shape</span>
-                    </div>,
-                  ]}
-                  icon={
-                    <img
-                      alt={s.action ? s.action!.name : s.name}
-                      src={getStepIcon(process.env.PUBLIC_URL, s)}
-                      width={24}
-                      height={24}
-                    />
-                  }
-                  actions={
-                    <>
-                      <ButtonLink href={this.props.configureStepHref(idx, s)}>
-                        Configure
-                      </ButtonLink>
-                      <ButtonLink href={'#'} as={'danger'}>
-                        <i className="fa fa-trash" />
-                      </ButtonLink>
-                    </>
-                  }
-                />
-                {idx < this.props.steps.length - 1 && (
-                  <IntegrationFlowAddStep
-                    active={false}
-                    showDetails={false}
-                    addStepHref={this.props.addStepHref(idx + 1)}
-                    i18nAddStep={'Add a step'}
+      <Translation ns={['integrations', 'shared']}>
+        {t => (
+          <WithIntegrationHelpers>
+            {({ removeStep }) => (
+              <>
+                {this.state.showDeleteDialog && (
+                  <ConfirmationDialog
+                    buttonStyle={ConfirmationButtonStyle.NORMAL}
+                    icon={ConfirmationIconType.DANGER}
+                    i18nCancelButtonText={t('shared:Cancel')}
+                    i18nConfirmButtonText={t('shared:Delete')}
+                    i18nConfirmationMessage={t(
+                      'integrations:editor:confirmDeleteStepDialogBody'
+                    )}
+                    i18nTitle={t(
+                      'integrations:editor:confirmDeleteStepDialogTitle'
+                    )}
+                    showDialog={this.state.showDeleteDialog}
+                    onCancel={this.closeDeleteDialog}
+                    onConfirm={() => {
+                      this.handleDeleteConfirm();
+                      removeStepFromFlow(
```

Comment:
yeah, this here function returns a copy of the integration object with that step removed from the indicated flow.  This is where I guess we need to figure out how to get the page to see the return value of this function.

Reply:
Just to confirm then, what you are saying is that if I set that function as a local var, it should return one less step than the original integration, correct? But I do this:
```
onConfirm={() => {
                      console.log('oldInt: ' + JSON.stringify(this.props.integration));
                      this.handleDeleteConfirm();
                      const newInt = removeStepFromFlow(
                        this.props.integration!,
                        this.props.flowId!,
                        this.state.position!
                      );
                      console.log('newInt: ' + JSON.stringify(newInt));
                    }}
```
And the old integration's `steps` array count is the same as the new one. I would expect one less step for the returned copy of the integration object. I tested that the data passed is correct, so I want to make sure the function is working as expected.

Reply:
Got it, I tested both `const steps = [...flow!.steps!];` (current implementation) and `const steps = getSteps(integration, flowId);` (my changes) for the steps, and it seems to work with my changes. Moving to the next step now (updating the page).